builder/builder-next/worker/worker.go

```diff
@@ -160,10 +160,10 @@ func (w *Worker) Platforms(noCache bool) []ocispec.Platform {
 	if noCache {
 		pm := make(map[string]struct{}, len(w.Opt.Platforms))
 		for _, p := range w.Opt.Platforms {
-			pm[platforms.Format(p)] = struct{}{}
+			pm[platforms.FormatAll(p)] = struct{}{}
```

Comment:
This should be updated like https://github.com/moby/buildkit/blob/master/worker/base/worker.go#L264

Reply:
Thanks; I get what you are recommending, but that seems outside the scope of this PR (i.e., you are suggesting a refactoring of the entire function, where as this PR is simply about replacing `Format()` with `FormatAll()`). 

Can we do what you are suggesting in a follow-up PR?

Reply:
Then don't make changes to this part of the code.

Reply:
Just so I understand: is that because the change I am proposing incorrect? If so where, I don't see the error.

Reply:
The old code is incorrect and the new code is incorrect as well.

Reply:
I assume the change in [builder/builder-next/adapters/containerimage/pull.go](https://github.com/moby/moby/pull/49973/files#diff-2b45354ea4099a5c500a68697fcdf8ab51fa9c052b457dcd125f72c72c50781b) looks good to you?

Reply:
Looks like that code was originally added in https://github.com/moby/moby/commit/dbffbe8f9bbeb784a5ebbd8979e548370548b6e8 (https://github.com/moby/moby/pull/41601); was this incorrect all this time? 😓

Reply:
Oh, never mind; I looked at the wrong part of the diff 🙈 ignore me.